src/Data/Lens/Index.purs

```diff
@@ -15,6 +15,23 @@ import Data.Set as S
 import Data.StrMap as SM
 import Data.Traversable (traverse)
 
+-- | `Index` is useful when a focus element might not be present.
+-- | It allows you to `set` an existing focus element, but you cannot add a
+-- | new one or delete an old one.
```

Comment:
I'm not sure about this explanation, it sounds more like a description of partial lenses (prism, traversal, etc) to me in general than being specific to `Index` - it makes no mention really of what the `Index` class is for.

I don't have any good suggestions for this one though, I don't find myself having a need for `Index` or `At` very commonly, so I'm not sure if there are good examples outside of saying they're for things like arrays and maps.

It might also be worth contrasting `Index` and `At` in the description of both?

Reply:
Changed explanation.

Didn't say anything about the type class because, it seems to me that the meaning of `Index` (and `At`) really only matters if you're making a new instance. (Note how neither appears When you're using the `Lens` or `Traversal` type aliases.) And making a new instance seems  more something to put in a book. And -- what a coincidence -- there's some of that in this book: https://leanpub.com/lenses

I've added references to each other in both `At` and `Index`.